_docs/rfc/imports.md

```diff
@@ -0,0 +1,766 @@
+
+Cons:
+
+- Encourages monkey patching, which generally increases complexity and deteriorates readability/maintainability.
+- Implementation can be complex due to graph logic between globals and locals to allow reuse across the two.
+- Does not solve all the problems in this RFC. E.g `globals` does not address the need for multiple includes and
+  fine grained control over merging.
+
+Ultimately, the complexity of the implementation and the monkey patching behavior suggested that it may be better to
+look for an alternative implementation.
+
+
+### hcldecode or read_terragrunt_config helper function
+
+Instead of defining a dedicated block for `import`, we could define a helper function that parses the relevant config.
+For example, the explicit triple import example in the [hierarchical variables use
+case](#hierarchical-variables-included-across-multiple-terragrunt-hcl-files) can be implemented as:
+
+```
+locals {
+  root_config = read_terragrunt_config("../../../root.hcl")
+  region_config = read_terragrunt_config("../../region.hcl")
+  env_config = read_terragrunt_config("../env.hcl")
+}
+
+inputs = merge(
+  local.root_config.inputs,
+  local.region_config.inputs,
+  local.env_config.inputs,
+  {
+    # args to module
+  },
+)
+```
+
+Pros:
+
+- Relatively simple implementation. It is significantly easier to add helper functions than it is to introduce new
+  blocks in terragrunt.
+- Can address all the same use cases as `import` blocks.
+
+Cons:
+
+- Everything has to be explicit in the config. We can not support automatic merging capabilities when using helper
+  functions, as you can't manipulate the config in a helper. This can lead to very verbose configurations if one wants
+  to use this in place of `include`.
+
+Ultimately, the verbosity that this implementation entails would encourage users to continue to use `include`. While
+this is not a bad thing, it was preferred to reduce constructs instead of introducing new ones. Note that while `import`
+introduces a new construct, the proposal includes deprecating the `include` block in favor of `import` such that
+eventually `import` replaces `include`.
+
+
+## Open Question
+
+- While a major design goal in this RFC is to keep the principles of explicit over implicit, practically speaking it may
+  not be feasible to achieve that design goal to support all the use cases of the community. In that regard, it may be
+  useful to implement a minimal set of features that don't take a whole lot of cognitive load to understand. Should the
+  following functions be implemented for use with `import`?
+    - `path_relative_to_import` and `path_relative_from_import`: These are the equivalent functions of the ones named
+      for `include`.
+    - `find_in_parent_folders_from_importing_config`: This function is the version of `find_in_parent_folders` that
+      works in the context of the config that is importing the current config.
```

Comment:
Use case that would benefit heavily from this: https://github.com/gruntwork-io/terragrunt/issues/744#issuecomment-580092107